wavesrv/pkg/remote/openai/openai.go

```diff
@@ -68,6 +74,96 @@ func RunCompletion(ctx context.Context, opts *sstore.OpenAIOptsType, prompt []ss
 	return marshalResponse(apiResp), nil
 }
 
+func RunCloudCompletion(ctx context.Context, opts *sstore.OpenAIOptsType, prompt []sstore.OpenAIPromptMessageType) ([]*packet.OpenAIPacketType, error) {
+	if opts == nil {
+		return nil, fmt.Errorf("no openai opts found")
+	}
+	if opts.Model == "" {
+		return nil, fmt.Errorf("no openai model specified")
+	}
+	cloudCompletionRequestConfig := sstore.OpenAICloudCompletionRequest{
+		Model:      opts.Model,
+		Prompt:     prompt,
+		MaxTokens:  opts.MaxTokens,
+		MaxChoices: opts.MaxChoices,
+	}
+	const cloudTestAddr = "http://127.0.0.1:7999/api/chat-completion"
+	payloadBuf := new(bytes.Buffer)
+	json.NewEncoder(payloadBuf).Encode(cloudCompletionRequestConfig)
+	httpreq, err := http.NewRequest("POST", cloudTestAddr, payloadBuf)
+	if err != nil {
+		return nil, fmt.Errorf("request create err: %v", err)
+	}
+	httpresp, err := http.DefaultClient.Do(httpreq)
+	if err != nil {
+		return nil, fmt.Errorf("request err: %v", err)
+	}
+	defer httpresp.Body.Close()
+	body, err := io.ReadAll(httpresp.Body)
+	var apiResp openaiapi.ChatCompletionResponse
+	type jsonResponse struct {
+		Data openaiapi.ChatCompletionResponse
+	}
+	var httpjsonresp jsonResponse
+	err = json.Unmarshal(body, &httpjsonresp)
+	if err != nil {
+		return nil, fmt.Errorf("error decoding json output %v", err)
+	}
+	apiResp = httpjsonresp.Data
+	return marshalResponse(apiResp), nil
+}
+
+func RunCloudCompletionStream(ctx context.Context, opts *sstore.OpenAIOptsType, prompt []sstore.OpenAIPromptMessageType) (chan *packet.OpenAIPacketType, error) {
+	const AWSLambdaCentralWSAddr = "wss://5lfzlg5crl.execute-api.us-west-2.amazonaws.com/dev/"
+	if opts == nil {
+		return nil, fmt.Errorf("no openai opts found")
+	}
+	if opts.Model == "" {
+		return nil, fmt.Errorf("no openai model specified")
+	}
+	conn, _, err := websocket.DefaultDialer.Dial(AWSLambdaCentralWSAddr, nil)
+	if err != nil {
+		log.Printf("Websocket error: %v", err)
+		return nil, fmt.Errorf("Websocket error: %v", err)
+	}
+	cloudCompletionRequestConfig := sstore.OpenAICloudCompletionRequest{
+		Model:      opts.Model,
+		Prompt:     prompt,
+		MaxTokens:  opts.MaxTokens,
+		MaxChoices: opts.MaxChoices,
+	}
+	configMessageBuf := new(bytes.Buffer)
```

Comment:
should I change this to marshal instead of decode?

Reply:
ya, it is easier that way.  Marshal will give you a []byte directly, then you don't even need the bytes.Buffer.

Reply:
addressed